pints/toy/_distributions.py

```diff
@@ -0,0 +1,186 @@
+            cor = 0.5 * np.ones((self._dimension, self._dimension))
+            np.fill_diagonal(cor, 1)
+            cov = cor
+            for i in range(0, self._dimension):
+                for j in range(0, self._dimension):
+                    cov[i, j] *= np.sqrt(i + 1) * np.sqrt(j + 1)
+        self._cov = cov
+        if mu is None:
+            mu = np.repeat(0, self._dimension)
+        self._mu = mu
+
+    def __call__(self, x):
+        if len(x) != self._dimension:
+            raise ValueError('Dimensions of x must equal pdf dimensions.')
+        log_pdf = multivariate_normal.logpdf(x, mean=self._mu, cov=self._cov)
+        return log_pdf
+
+    def dimension(self):
+        """ See: :meth:`LogPDF.dimension()`. """
+        return self._dimension
+
+
+class BimodalMultivariateGaussianLogPDF(pints.LogPDF):
+    """
+    *Extends:* :class:`LogPDF`.
+
+    Bimodal multivariate (un-normalised) Gaussian. Default is 2D with modes
+    at ``(0, 0)`` and ``(10, 10)`` with independent unit covariance matrices.
+    """
+    def __init__(self, dimension=2, mu1=[0, 0], mu2=[10, 10],
+                 cov1=None, cov2=None):
+        self._dimension = int(dimension)
+        if len(mu1) != self._dimension:
+            raise ValueError(
+                'Length of mean must equal pdf dimensions (default is 2).')
+        if len(mu2) != self._dimension:
+            raise ValueError(
+                'Length of mean must equal pdf dimensions (default is 2).')
+        if cov1 is None:
+            cov1 = np.diag(np.repeat(1, self._dimension))
+        else:
+            dim1, dim2 = cov1.shape
+            if dim1 != dim2:
+                raise ValueError('Covariance matrix must be square')
+            if not dim1 == dim2 == self._dimension:
+                raise ValueError(
+                    'Dimensions of covariance matrix must match pdf dimensions'
+                    ' (default is 2).')
+        if cov2 is None:
+            cov1 = np.diag(np.repeat(1, self._dimension))
+        else:
+            dim1, dim2 = cov2.shape
+            if dim1 != dim2:
+                raise ValueError('Covariance matrix must be square')
+            if not dim1 == dim2 == self._dimension:
+                raise ValueError(
+                    'Dimensions of covariance matrix must match pdf dimensions'
+                    ' (default is 2).')
+        self._mu1 = mu1
+        self._mu2 = mu2
+        self._cov1 = cov1
+        self._cov2 = cov2
+
+    def __call__(self, x):
+        if len(x) != self._dimension:
+            raise ValueError('Dimensions of x must equal pdf dimensions.')
+        log_pdf = np.log(
+            + multivariate_normal.pdf(x, mean=self._mu1, cov=self._cov1)
+            + multivariate_normal.pdf(x, mean=self._mu2, cov=self._cov2))
+        return log_pdf
+
+    def dimension(self):
+        """ See: :meth:`LogPDF.dimension()`. """
+        return self._dimension
+
+
+class RosenbrockLogPDF(pints.LogPDF):
```

Comment:
Needs:

- [x] An example in a notebook
- [ ] A test
- [ ] Sphinx documentation